src/models/generic/WorkflowModel.ts

```diff
@@ -1029,4 +1052,176 @@ export abstract class WorkflowModel extends ValidationBase implements Serializab
         new UnimplementedMethodException("getSourceConnectionId");
         return undefined;
     }
+
+    /**
+     * Validate all visible connections and sets correct validity state on destinations
+     */
+    private validateConnections(): Promise<any> {
+
+        const sources = this.gatherSources();
+        const destinations = this.gatherDestinations();
+
+        this.connections.forEach((connection) => {
+
+            if (connection.isVisible) {
+                const source = sources.find((item) => {
+                    return item.connectionId === connection.source.id;
+                });
+
+                const destination = destinations.find((item) => {
+                    return item.connectionId === connection.destination.id;
+                });
+
+                this.validateConnection(destination, source);
+            }
+        });
+
+        return new Promise((resolve) => resolve());
+    }
+
+    /**
+     * Validate connection between source and destination and sets correct validity state on destination
+     */
+    private validateConnection (destination: WorkflowOutputParameterModel | WorkflowStepInputModel,
+                        source: WorkflowInputParameterModel | WorkflowStepOutputModel, updateConnection = false) {
+
+        if (!source || !destination) {
+            return;
+        }
+
+        let isValid = true;
+
+        const destinationType = destination.type;
+        const sourceType = source.type;
+
+        // Check file types, true if at least one source file type is in destination file types
+        const checkFileTypes = () => {
+            if (source.fileTypes.length && !source.fileTypes
```

Comment:
Also, check that `destination.fileTypes.length` is greater than 0, because in case the destination doesn't have defined fileTypes, we can't know if it's a mismatch.